src/runtime/story.rs

```diff
@@ -0,0 +1,192 @@
+use std::{
+    borrow::Borrow,
+    fmt::{self, Write},
+    rc::Rc,
+};
+
+use crate::runtime_graph::RuntimeGraph;
+
+use super::{
```

Comment:
for consistency, can we have all the imports be `use crate::` when applicable :)
I'll see about adding a rustfmt option for that